### PR TITLE
Tar: Correct and test modification time

### PR DESCRIPTION
**Pull request by Geod24**
Best reviewed commit by commit (the first commit only removes trailing whitespace, it's fairly trivial).
I had an issue when trying to use this library with files downloaded from the internet (I wanted to edit a file within the archive).
This fixes it for me.

### CODE REVIEW

File: source/archive/tar.d [hunk 1]
```diff
@@ -620,22 +622,30 @@ unittest
...
-    
+
     // Add a directory that does not exist.
     output.addDirectory("newdirectory/");
 
     // Remove unused directories
     output.removeEmptyDirectories();
-    
+
+<<<<<<< HEAD
```
Comment (WebFreak001) [inline source/archive/tar.d:643]: merge fail

**Review by WebFreak001 — CHANGES_REQUESTED** [review]
are the whitespace changes really needed in this PR?

**Comment by Geod24** [comment]
> are the whitespace changes really needed in this PR?

Well there is one line where the diff overlaps, so I figured I'd submit it alongside.
Plus it's in a separate commit, so a rebase would help. But since you have merge access (and are active), I can split it in a different PR if you'd like.

File: source/archive/tar.d [hunk 2]
```diff
@@ -560,7 +561,7 @@ public class TarPolicy
                 header.ownerId = rightJustify(intToOctalStr(0), 7) ~ "\0";
                 header.groupId = rightJustify(intToOctalStr(0), 7) ~ "\0";
                 header.size = rightJustify(intToOctalStr(0), 11) ~ " ";
-                header.modificationTime = rightJustify(longToOctalStr(directory.modificationTime), 11) ~ " ";
```
Comment (WebFreak001) [inline source/archive/tar.d:563]: you removed the 11 padding here, which works for times after 2004, but what about older ones that aren't 11 octal digits? I don't really know how the tar format works internally, but I assume the 11 was here for a reason for small dates before.

Comment (Geod24) [reply source/archive/tar.d:563]: It's 11 chars, but the padding is usually 0 with newer program (the field is nullified earlier in the code).
So what this does essentially is replace space padding with `\0` padding.

**Review by WebFreak001 — APPROVED** [review]

**Pull request merged**